app/controllers/App.scala

```diff
@@ -1,14 +1,27 @@
 package controllers
 
+import cats.syntax.either._
 import config.Config
+import database.MetricsDB
 import play.api.Logger
 import play.api.libs.ws.WSClient
 import play.api.mvc._
+import util.Utils._
+import util.Parser._
 
-class App(val wsClient: WSClient) extends Controller with PanDomainAuthActions {
+class App(val wsClient: WSClient, val config: Config, val db: MetricsDB) extends Controller with PanDomainAuthActions {
 
   def index = AuthAction {
-    Logger.info(s"I am the ${Config.appName}")
+    Logger.info(s"I am the ${config.appName}")
     Ok(views.html.index())
   }
+
+  def createdInCopyContent = AuthAction { req =>
+    val x = for {
+      body <- extractRequestBody(req.body.asJson.map(_.toString))
+      json <- stringToJson(body)
+      data <- jsonToInCopyData(json)
+    } yield data
```

Comment:
This is too clever for me! I have tried to work out what we are doing here but I can't.

Are we just running this for the side-effect of throwing an exception if it is not valid?

Could we make that more explicit by just doing something like:

```scala
for {
  body <- extractRequestBody(req.body.asJson.map(_.toString))
  json <- stringToJson(body)
  _  <- jsonToInCopyData(json)
} yield ()
```

as a side note, I can't figure out why this won't compile if I change it to:

```scala
for {
  body <- extractRequestBody(req.body.asJson.map(_.toString))
  json <- stringToJson(body)
  _  <- jsonToInCopyData(json)
}
```

I would love to know why if you know!